src/main/java/Duke.java

```diff
@@ -1,10 +1,189 @@
+
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.stream.*;
+
+
 public class Duke {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void talk() {
+
+        java.nio.file.Path dirpath = Paths.get("data");
+        Path taskListPath = Paths.get("data", "DukeTask.txt");
+        boolean directoryExists = Files.exists(dirpath);
+        boolean fileExist = Files.exists(taskListPath);
+        if(!directoryExists) {
+            try{
+                Files.createDirectory(dirpath);
+            } catch (IOException e) {
+                System.out.println(e.getMessage());
```

Comment:
Maybe can write these functions as DukeExceptions instead, so that all these errors can be abstracted for a method "printDukeExceptionMessage".